File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let result = 0;
    
    dfs(0, 0);
    function dfs(idx, sum){
        if(idx === numbers.length){
            if(sum === target) result++;
            return;
        }
        
        dfs(idx + 1, sum + numbers[idx]);
        dfs(idx + 1, sum - numbers[idx]);
    }
    
    return result;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    result = 0

    def dfs(idx, sum):
        nonlocal result
        if idx == len(numbers):
            if sum == target:
                result += 1
            return
        
        dfs(idx+1, sum + numbers[idx])
        dfs(idx+1, sum - numbers[idx])
    
    dfs(0, 0)
    return result
